core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/source/sql/asterixdb/AsterixDBSourceOpExec.scala

```diff
@@ -201,15 +205,49 @@ class AsterixDBSourceOpExec private[asterixdb] (
     * @throws IllegalArgumentException if attribute does not support string based search
     */
   @throws[IllegalArgumentException]
-  def addKeywordSearch(queryBuilder: StringBuilder): Unit = {
+  def addFilterConditions(queryBuilder: StringBuilder): Unit = {
+    if (search.getOrElse(false) && searchByColumn.isDefined && keywords.isDefined) {
+      val columnType = schema.getAttribute(searchByColumn.get).getType
 
-    val columnType = schema.getAttribute(searchByColumn.get).getType
+      if (columnType == AttributeType.STRING) {
 
-    if (columnType == AttributeType.STRING) {
+        queryBuilder ++= " AND ftcontains(" + searchByColumn.get + ", " + keywords.get + ") "
+      } else
+        throw new IllegalArgumentException("Can't do keyword search on type " + columnType.toString)
+    }
+
+    if (searchByColumnForRegex.isDefined && regex.isDefined) {
+      val regexColumnType = schema.getAttribute(searchByColumnForRegex.get).getType
+      if (regexColumnType == AttributeType.STRING) {
+        queryBuilder ++= "AND regexp_contains(" + searchByColumnForRegex.get + ", \"" + regex.get + "\") "
+      } else
+        throw new IllegalArgumentException(
+          "Can't do regex search on type " + regexColumnType.toString
+        )
+    }
 
-      queryBuilder ++= " AND ftcontains(" + searchByColumn.get + ", " + keywords.get + ") "
-    } else
-      throw new IllegalArgumentException("Can't do keyword search on type " + columnType.toString)
+    // geolocation must contain more than 1 points to from a rectangle or polygon
+    if (geolocation.size > 1 && geoAttributes.nonEmpty) {
+      val shape = {
+        val points = geolocation.flatMap(s => s.split(",").map(sub => sub.toDouble))
+        if (geolocation.size == 2) {
+          "create_rectangle(create_point(%.6f,%.6f), create_point(%.6f,%.6f))".format(points: _*)
+        } else {
+          "create_polygon([" + points.map(x => "%.6f".format(x)).mkString(",") + "])"
+        }
```

Comment:
wouldn't this have some parse error? I think we need to handle invalid inputs

Reply:
If we have a parse error, should it be thrown automatically? Do we have to catch it and re-throw it again?

Reply:
the question is simple: if user enters invalid input, would frontend see error in order to update it?

Reply:
If the above throw statement can be seen by the frontend, then yes.